staging/src/k8s.io/code-generator/cmd/validation-gen/validation.go

```diff
@@ -943,6 +974,36 @@ func (g *genValidations) emitValidationForChild(c *generator.Context, thisChild
 				case types.Struct:
 					// Call the type's validation function.
 					g.emitCallToOtherTypeFunc(c, fld.node, bufsw)
+					for _, subchild := range fld.inner {
+						if len(subchild.name) == 0 {
+							panic(fmt.Sprintf("missing child name for field in %v", thisNode))
+						}
+						if len(subchild.jsonName) == 0 {
+							panic(fmt.Sprintf("missing child JSON name for field %v.%s", thisNode, subchild.name))
+						}
+
+						leafType, typePfx, exprPfx := getLeafTypeAndPrefixes(subchild.childType)
+						targs := targs.WithArgs(generator.Args{
+							"inType":       fld.childType,
+							"fieldName":    subchild.name,
+							"fieldJSON":    subchild.jsonName,
+							"fieldType":    leafType,
+							"fieldTypePfx": typePfx,
+							"fieldExprPfx": exprPfx,
+						})
+						bufsw.Do("// field $.inType|raw$.$.fieldName$\n", targs)
+						bufsw.Do("errs = append(errs,\n", targs)
+						bufsw.Do("  func(obj, oldObj $.fieldTypePfx$$.fieldType|raw$, fldPath *$.field.Path|raw$) (errs $.field.ErrorList|raw$) {\n", targs)
+
+						if !subchild.fieldValidations.Empty() {
```

Comment:
If we get here and find an empty list of field validations, it would be very weird, right?  These children were only added BECAUSE they have inner validation.

Maybe panic here is we find it empty - sort of an "assert".  We do it in a few other places where we REALLY do not expect a bad input.

Reply:
Done, removed that check and changed the logic to be:

```go
						if subchild.fieldValidations.Empty() {
							panic(fmt.Sprintf("found non-empty field validations in node.inner for node: %v", subchild))
						}

						emitCallsToValidators(c, subchild.fieldValidations.Functions, bufsw)f
```